onnxruntime/core/providers/cuda/gpu_data_transfer.cc

```diff
@@ -32,8 +32,14 @@ common::Status GPUDataTransfer::CopyTensor(const Tensor& src, Tensor& dst, int e
 
   if (dst_device.Type() == OrtDevice::GPU) {
     if (src_device.Type() == OrtDevice::CPU && src_device.MemType() == OrtDevice::MemType::CUDA_PINNED) {
-      // copy from pinned memory to GPU, this is non-blocking
+      // copy from pinned memory to GPU
+      // this is blocking for Windows as it's not thread-safe with Async call - ReduceMeanGrad UT with RelWithDebInfo build
+      // is broken consistently 
+#ifdef _WIN32
+      CUDA_RETURN_IF_ERROR(cudaMemcpy(dst_data, src_data, bytes, cudaMemcpyHostToDevice));
+#else
```

Comment:
Will Size op trigger this path?

Reply:
yes - now reverted and make proper fix after discussing with Ke.

Reply:
yes, to me this is a workaround hiding some issue in ORT.